src/dbt/kipptaf/models/reporting/base/base_reporting__promotional_status.sql

```diff
@@ -0,0 +1,362 @@
+with
+    attendance as (
+        select
+            mem._dbt_source_relation,
+            mem.studentid,
+            mem.yearid,
+            round(avg(mem.attendancevalue), 2) as ada_term_running,
+            coalesce(sum(abs(mem.attendancevalue - 1)), 0) as n_absences_y1_running,
+
+            rt.name as term,
+        from {{ ref("int_powerschool__ps_adaadm_daily_ctod") }} as mem
+        inner join
+            {{ ref("stg_reporting__terms") }} as rt
+            on mem.calendardate <= rt.end_date
+            and mem.schoolid = rt.school_id
+            and mem.yearid = rt.powerschool_year_id
+            and rt.type = 'RT'
+        where
+            mem.membershipvalue = 1
+            and mem.calendardate <= current_date('America/New_York')
+        group by mem._dbt_source_relation, mem.yearid, mem.studentid, rt.name
+    ),
+
+    credits as (
+        select
+            _dbt_source_relation,
+            studentid,
+            academic_year,
+            storecode,
+            sum(potential_credit_hours) as enrolled_credit_hours,
+            sum(if(y1_letter_grade_adjusted in ('F', 'F*'), 1, 0)) as n_failing,
+            sum(
+                if(
+                    y1_letter_grade_adjusted not in ('F', 'F*')
+                    and y1_letter_grade_adjusted is not null,
+                    potential_credit_hours,
+                    null
+                )
+            ) as projected_credits_y1_term,
+        from {{ ref("base_powerschool__final_grades") }}
+        group by _dbt_source_relation, studentid, academic_year, storecode
+    ),
+
+    iready_dr as (
+        select
+            student_id,
+            academic_year_int,
+            subject,
+            most_recent_overall_relative_placement,
+        from {{ ref("base_iready__diagnostic_results") }}
+    ),
+
+    iready as (
+        select student_id, academic_year_int, iready_reading_recent, iready_math_recent,
+        from
+            iready_dr pivot (
+                max(most_recent_overall_relative_placement) for subject
+                in ('Reading' as iready_reading_recent, 'Math' as iready_math_recent)
+            )
+    ),
+    promo as (
+        select
+            co.student_number,
+            co.lastfirst,
+            co.region,
+            co.school_level,
+            co.school_abbreviation,
+            co.grade_level,
+            co.advisory_name,
+            co.advisor_lastfirst as advisor_name,
+            co.lep_status,
+            co.ethnicity,
+            co.gender,
+            co.is_retained_year,
+            co.is_retained_ever,
+            case
+                when co.spedlep like 'SPED%' then 'Has IEP' else co.spedlep
+            end as iep_status,
```

Comment:
keep only cols necessary for subsequent calcs in this view and joins in downstream reporting views